src/converters/fromZigbee.ts

```diff
@@ -374,7 +374,13 @@ const converters1 = {
         type: ['attributeReport', 'readResponse'],
         convert: (model, msg, publish, options, meta) => {
             const payload: KeyValueAny = {};
-            if (msg.data.hasOwnProperty('batteryPercentageRemaining') && msg.data['batteryPercentageRemaining'] < 255) {
+            // If voltageToPercentage is specified, it means we do not trust the percentage
+            // returned by the device and are instead calculating it ourselves.
+            if (
+                !model.meta.battery.hasOwnProperty('voltageToPercentage') &&
```

Comment:
```suggestion
                model.meta?.battery?.voltageToPercentage != null &&
```

Reply:
TIL'ed about optional chaining. Thanks!

I'm writing a test for this now.